claim/schema.py

```diff
@@ -235,6 +241,15 @@ def resolve_claim_officers(self, info, search=None, **kwargs):
             )
         return qs
 
+    def resolve_fsp_from_claim(self, info, **kwargs):
+        if not info.context.user.has_perms(ClaimConfig.gql_query_claim_officers_perms):
+            raise PermissionDenied(_("unauthorized"))
+        result = Insuree.objects.filter(
+            chf_id=kwargs['insuree_code'],
+            *filter_validity(validity=kwargs['date_claimed']),
+        ).order_by(F('validity_to').asc(nulls_last=True)).first().health_facility
```

Comment:
why having orderby validity ?
```suggestion
        ).first().health_facility
```